src/components/Booklist.jsx

```diff
@@ -1,11 +1,16 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import Book from './Book';
 import Input from './Input';
+import { fetchData } from '../redux/books/books';
 
 import classes from './Booklist.module.css';
 
 const Booklist = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchData());
+  }, []);
```

Comment:
- [OPTIONAL] There are some errors that come up when I start the app. Please screenshot the errors and please try to handle them correctly.

Screenshot|
:--------------:|
![a1](https://user-images.githubusercontent.com/39523312/161048746-a89c162e-f84d-424d-a70f-42f5074e87a0.png)




- [OPTIONAL] I think you might need to work on the re-fetching every time someone navigates to the books component. Please note that you should fetch the data once and leave the management side to redux. Even if a visitor navigates multiple times, it should not re-fetch from the external API again. The re-fetching is causing your state to make copies of the current state on every render.

Please see the test demo to understand the point I am trying to make here.

Demo |
:-------:|
![a1](https://user-images.githubusercontent.com/39523312/161048272-e6df3280-d5cd-4b1b-8697-526aff185784.gif)

Reply:
Thanks, I noticed that after I submitted the pull request, and I already fixed it while you were reviewing the pull request. Regarding the errors, these might occur if you have prettier extension ON. But I assure you they don't appear after turning off any extension which auto formats.

Reply:
Good job in fixing them @muneebulrehman.